cpp/src/arrow/meson.build

```diff
@@ -293,7 +293,8 @@ if needs_csv
 endif
 
 if needs_json or needs_integration
-    rapidjson_dep = dependency('rapidjson', include_type: 'system')
+    rapidjson_proj = subproject('rapidjson')
+    rapidjson_dep = rapidjson_proj.get_variable('rapidjson_dep')
```

Comment:
Does this mean that we always use bundled RapidJSON?
Can we use bundled RapidJSON only when system RapidJSON is old?